aiohttp/web_urldispatcher.py

```diff
@@ -227,11 +227,29 @@ def set_current_app(self, app):
         finally:
             self._current_app = prev
 
+    def add_map(self, match_dict):
+        if self._frozen:
+            raise RuntimeError("Cannot change maps stack after .freeze() call")
+        self._variables.maps.append(match_dict)
```

Comment:
Let's consider the following:
```
async def handler(request):
    return web.Response(text=request.math_info['name'])

subapp = web.Application()
subapp.add_routes([web.get('/{name}', handler)])

app = web.Application()
app.add_subapp('/{name}', subapp)
```
What `name` should be returned by the handler?

Reply:
In this case, `request.match_info['name']` in the handler will return the matched value of the innermost `{name}` definition: the route one.

Reply:
Perfect!